src/main/java/common/db/DBPrefs.java

```diff
@@ -13,136 +13,166 @@
+ * @author - ahmetkucuk
+ * @author - kqian5
  */
 public class DBPrefs {
 
-//    public static final String DB_HOST = "postgres-exposed";
-//    public static final String DB_NAME = "isd";
-//    public static String DB_USERNAME = "postgres";
-//    public static String DB_USER_PASSWORD = "r3mot3p8sswo4d";
+	final static String DEFAULT_RABBIT_MQ_HOST = "rabbitmq";
+	final static int DEFAULT_RABBIT_MQ_PORT = 5672;
 
-    final static String DEFAULT_RABBIT_MQ_HOST = "rabbitmq";
-    final static int DEFAULT_RABBIT_MQ_PORT = 5672;
+	public static final int DB_PORT = 5432;
 
-    public static final int DB_PORT = 5432;
+	/**
+	 * Sets the POSTGRE database configurations in a BasicDataSource object and
+	 * returns it
+	 * @return - the BasicDataSource object to return
+	 */
+	public static BasicDataSource getDataSource() {
 
+		Map<String, String> env = System.getenv();
 
-    public static BasicDataSource getDataSource() {
-
-        Map<String, String> env = System.getenv();
-
-        BasicDataSource dbPoolSourc = new BasicDataSource();
+		BasicDataSource dbPoolSourc = new BasicDataSource();
+		dbPoolSourc.setSoftMinEvictableIdleTimeMillis(6500);
+		dbPoolSourc.setDefaultAutoCommit(true);
+		dbPoolSourc.setPoolPreparedStatements(false);
+		dbPoolSourc.setDefaultQueryTimeout(60);
+		dbPoolSourc.setMinIdle(1);
+		dbPoolSourc.setMaxIdle(10);
+		dbPoolSourc.setMaxTotal(100);
+		dbPoolSourc.setUsername(env.get("POSTGRES_USER"));
+		dbPoolSourc.setPassword(env.get("POSTGRES_PASSWORD"));
+		dbPoolSourc.setValidationQuery("SELECT 1;");
+		dbPoolSourc.setDriverClassName("org.postgresql.Driver");
+		dbPoolSourc.setUrl("jdbc:postgresql://" + env.get("POSTGRES_DB_HOST") + "/" + env.get("POSTGRES_DB"));
+		return dbPoolSourc;
+	}
+
+	/**
+	 * Method that creates a map of all the necessary configuration settings for a
+	 * postgres db
+	 * 
+	 * @return - map of setting to its value
+	 */
+	public static Map<String, String> getTestEnv() {
+		Map<String, String> env = new HashMap<>();
+		env.put("POSTGRES_USER", "postgres");
+		env.put("POSTGRES_PASSWORD", "");
+		env.put("POSTGRES_DB", "postgres");
+		env.put("POSTGRES_DB_HOST", "localhost");
+		return env;
+	}
+
+	/**
+	 * Method that pauses the thread for 2000 milliseconds
+	 */
+	public static void pause() {
+		try {
+			Thread.sleep(2000);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Method that attempts to connect to database until a connection is successful
+	 * and checks connection by perform sql command "SELECT 1"
+	 */
+	public static void waitUntilConnected() {
+
+		boolean connected = false;
+		Connection ex = null;
+		while (!connected) {
+			try {
+				DataSource dsourc = getDataSource();
+				try {
+					ex = dsourc.getConnection();
+					ex.setAutoCommit(true);
+					Statement ex1 = ex.createStatement();
+					connected = ex1.execute("SELECT 1;");
+				} catch (SQLException var7) {
+					System.out.println("Failed to Connect. Will retry");
```

Comment:
Can you have better error messages for all of these exception messages? Retry in X seconds etc.